com.unity.visualeffectgraph/Editor/Data/VFXDataParticle.cs

```diff
@@ -743,7 +743,9 @@ public bool NeedsGlobalSort()
                 var temporaryBufferMappings = new List<VFXMappingTemporary>();
 
                 var context = m_Contexts[i];
-                var contextData = contextToCompiledData[context];
+                //An output context can be skipped due to invalid current RP
+                if (!contextToCompiledData.TryGetValue(context, out var contextData))
+                    continue;
```

Comment:
This does not look correct. Shouldnt we use compilable contexts instead of m_Contexts through the whole method?

Reply:
Indeed, we should test it in InitImplicitContexts to prevent building useless passes, I will propose a more appropriate change.

Reply:
I'm testing this change hasn't any  ⬇️ 
[_filter_context.diff](https://github.com/Unity-Technologies/Graphics/files/6452605/_filter_context.log)
![_vfx_switch_pipeline_cleaner](https://user-images.githubusercontent.com/22494706/117671946-a02fab00-b1a9-11eb-833d-28afdf7e18a7.gif)
https://user-images.githubusercontent.com/22494706/117671949-a1f96e80-b1a9-11eb-984a-363d946b3fb7.mp4

Reply:
Update, It changes a bit the behavior of VFXAttributeTests : https://github.com/Unity-Technologies/Graphics/blob/c6ae7599b33ca48852eddd592b3e40f33f2dd982/TestProjects/VisualEffectGraph_HDRP/Assets/AllTests/Editor/Tests/VFXAttributeTests.cs#L73 ⏳